tennis-scheduler-web-app/src/components/Registration.jsx

```diff
@@ -0,0 +1,72 @@
+import React, { useState, useEffect } from "react"
+import { Link } from 'react-router-dom'
+
+export const Registration = () => {
+    const [formValues, setFormValues] = useState({ name: "", surname: "", email: "", password: "", confirmPassword: "" });
+    const [formErrors, setFormErrors] = useState({});
+    const [isSubmit, setIsSubmit] = useState(false);
+
+    const handleChange = e => {
+        const { name, value } = e.target;
+        setFormValues((formValues) =>({ ...formValues, [name]: value }));
+    };
+
+    const registrationSubmit = e => {
+        e.preventDefault();
+        setFormErrors(validation(formValues));
+        setIsSubmit(true);
+    };
+
+    const validation = form => {
+        const errors = {};
+        const regex = /^[^\s@]+@[^\s@]+\.[^\s@]{2,}$/i;
+        if (!form.name) {
+            errors.name = "Name is required!";
+        }
+        if (!form.surname) {
+            errors.surname = "Surname is required!"
+        }
+        if (!form.email) {
+            errors.email = "Email is required!"
+        } else if (!regex.test(form.email)) {
+            errors.email = "This is not a valid email format!";
+        }
+        if (!form.password) {
+            errors.password = "Password is required!"
+        } else if (form.password.length < 8) {
+            errors.password = "Password must have minimum 8 characters";
+        }
+        if (!form.confirmPassword) {
+            errors.confirmPassword = "Confirm password is required!"
+        } else if (form.confirmPassword != form.password) {
```

Comment:
Friendly reminder: JavaScript has strict and not-so-strict comparison:
read about the differences between != and !==, == and ===.